Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/PhysicsPicking.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Math/IntersectSegment.h>
+#include <AzFramework/Physics/ShapeConfiguration.h>
+#include <AzToolsFramework/Viewport/ViewportTypes.h>
+#include <EMotionFX/Source/TransformData.h>
+#include <EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/RenderPlugin/RenderOptions.h>
+#include <Editor/Plugins/Ragdoll/PhysicsPicking.h>
+#include <Editor/Plugins/SkeletonOutliner/SkeletonOutlinerBus.h>
+#include <Editor/SkeletonModel.h>
+
+namespace EMotionFX
+{
+    struct PickingIntersection
+    {
+        bool m_intersected = false;
+        float m_distance = AZ::Constants::FloatMax;
+        size_t m_jointIndex = 0;
+    };
+
+    void CompareIntersection(PickingIntersection& closestIntersection, float distance, size_t jointIndex)
+    {
+        if (distance < closestIntersection.m_distance)
+        {
+            closestIntersection.m_intersected = true;
+            closestIntersection.m_distance = distance;
+            closestIntersection.m_jointIndex = jointIndex;
+        }
+    }
+
+    bool PhysicsPicking::HandleMouseInteraction(
+        [[maybe_unused]] const AzToolsFramework::ViewportInteraction::MouseInteractionEvent& mouseInteractionEvent)
+    {
+        const auto& mouseInteraction = mouseInteractionEvent.m_mouseInteraction;
+        if (!mouseInteraction.m_mouseButtons.Left() ||
+            mouseInteractionEvent.m_mouseEvent != AzToolsFramework::ViewportInteraction::MouseEvent::Down)
+        {
+            return false;
+        }
+
+        SkeletonModel* skeletonModel = nullptr;
+        SkeletonOutlinerRequestBus::BroadcastResult(skeletonModel, &SkeletonOutlinerRequests::GetModel);
+        if (!skeletonModel)
+        {
+            return false;
+        }
+
+        ActorInstance* actorInstance = skeletonModel->GetActorInstance();
+        if (!actorInstance)
+        {
+            return false;
+        }
+
+        PickingIntersection closestIntersection;
+        const EMotionFX::Skeleton* skeleton = actorInstance->GetActor()->GetSkeleton();
+        const AZ::Vector3& rayOrigin = mouseInteraction.m_mousePick.m_rayOrigin;
+        const AZ::Vector3& rayDirection = mouseInteraction.m_mousePick.m_rayDirection;
+
+        if (AZ::RHI::CheckBitsAny(m_renderFlags, EMotionFX::ActorRenderFlags::LineSkeleton))
```

Comment:
nit: maybe all the code inside this if could be moved to a function?

Reply:
Good call, moved this and the ragdoll collider intersection checking to separate functions.